### DISCUSSION TITLE
How can i use content negotiation when validate data?

### DISCUSSION THREAD

**Discussion by nandordudas** [adonisjs/core#1269] [Help] [answered]
Im using windows 10 with wsl 2 and just start project with `yarn start` - node `v14.5.0`.

Im using insomnia, but its uses this part of code.

```bash
#!/usr/env bash

curl -X POST localhost:3333 \
  -H 'accept: application/vnd.api+json' -H 'content-type: application/json' -d '{"username":""}'
```

```js
Route.post('/', async ({ request }) => (await request.validate(class {
  public schema = schema.create({username: schema.string()})
})))
// {"errors": [{"rule": "required", "field": "username", "message": "required validation failed"}]}
```

But, if accept part is hardcoded then it works.

```js
Route.post('/', async ({ request }) => {
  request.request.headers.accept = 'application/vnd.api+json'
  await request.validate(class {
    public schema = schema.create({username: schema.string()})
  })
})
// {"errors": [{"code": "required", "source": {"pointer": "username"}, "title": "required validation failed"}]}
```

What is im doing wrong?

**Comment by RomainLanz** [comment, marked as answer]
Hey @nandordudas! 👋 

If you want to have JSON API format back you need to send the `Accept` header everytime.
You can also force the `Accept` in your configuration `config/app.ts`.

Note that you are using the validator wrong.
You shouldn't pass an anonymous class.

https://preview.adonisjs.com/guides/validator/usage

**Comment by nandordudas** [comment]
> You can also force the `Accept` in your configuration `config/app.ts`.

Thanks!

> You shouldn't pass an anonymous class.

Yes, **this is** a very **bad example**! Only for demo, **no one use it** 😃
